FV3/gfsphysics/GFS_layer/GFS_physics_driver.F90

```diff
@@ -4486,6 +4503,44 @@ subroutine GFS_physics_driver                                 &
                               psautco_l, prautco_l, Model%evpco, Model%wminco, &
                               Tbd%phy_f3d(1,1,ntot3d-2), lprnt, ipr)
           else
+            #ifdef ENABLE_CALL_PY_FORT
+            ! copy tracer fields for state saving
```

Comment:
Sorry I don't have example results handy, but when I tried to use portions of arrays in the set_state operation (e.g., `set_state("cloud_water_mixing_ratio_output", Stateout%gq0(1,1,ntcw)`) it hasn't worked as I thought it would. I don't know whether this is a limitation of the fortran or call_py_fort, or if I'm using the wrong syntax.

Reply:
Thanks for the context. I saw you tried it that way. `set_state` uses assumed shaped arrays so I guess this is expected. Maybe leave a comment since it was a bit confusing?